### ISSUE TITLE
Add support for "Default Programs" panel

### ISSUE THREAD

**Issue by devcompl** [d2phap/ImageGlass#129]
Currently the only way (on windows 10 at least) to set ImageGlass as default image viewer is to manually change each associations for supported file extensions.
The ideal way is to add ImageGlass to "Default Programs" panel, this provides users with easy management of associations.

[Here](https://msdn.microsoft.com/en-us/library/windows/desktop/cc144154%28v=vs.85%29.aspx) is a microsoft guide about "Default Programs".

Basically the installer needs to write some registry keys.

**Comment by lovebdsobuj** [comment]
@devcompl, 
> Basically the installer needs to write some registry keys.

Yeah, I was try to write some registry for ImageGlass manually.
But I didn't get `ImageGlass` app in Default Programs list for associations Default Image viewer software.

Like these,

**ImageGlass.reg**
```
Windows Registry Editor Version 5.00

[HKEY_LOCAL_MACHINE\SOFTWARE\ImageGlass]
"ProgramFolder"="C:\\Program Files\\ImageGlass"
"ProgramPath"="C:\\Program Files\\ImageGlass\\ImageGlass.exe"

[HKEY_LOCAL_MACHINE\SOFTWARE\ImageGlass\Capabilities]
"ApplicationDescription"="A lightweight, versatile image viewer"
"ApplicationName"="ImageGlass"
```

**ImageGlass Capabilities.reg**
```
[HKEY_LOCAL_MACHINE\SOFTWARE\ImageGlass\Capabilities\FileAssociations]
".jpg"="ImageGlass.jpg"
```

**ImageGlass.jpg.reg**
```
Windows Registry Editor Version 5.00

[HKEY_LOCAL_MACHINE\SOFTWARE\Classes\ImageGlass.png]
@="Image (png) File"

[HKEY_LOCAL_MACHINE\SOFTWARE\Classes\ImageGlass.png\DefaultIcon]
@="C:\\Program Files\\ImageGlass\\icons\\png.ico"

[HKEY_LOCAL_MACHINE\SOFTWARE\Classes\ImageGlass.png\Shell]

[HKEY_LOCAL_MACHINE\SOFTWARE\Classes\ImageGlass.png\Shell\Open]
"FriendlyAppName"="ImageGlass"

[HKEY_LOCAL_MACHINE\SOFTWARE\Classes\ImageGlass.png\Shell\Open\Command]
@="\"C:\\Program Files\\ImageGlass\\ImageGlass.exe\" \"%1\""
```

**Comment by devcompl** [comment]
Is this going to be fixed in the next release?
Or maybe you could re-release the current version with fixed installer?

**Comment by d2phap** [comment]
Hi @devcompl , @lovebdsobuj 
I'm working on this issue.

Can you try importing this and let me know if it works?

**ImageGlass.Capabilities.reg**
```
Windows Registry Editor Version 5.00

[HKEY_LOCAL_MACHINE\SOFTWARE\PhapSoftware]

[HKEY_LOCAL_MACHINE\SOFTWARE\PhapSoftware\ImageGlass]

[HKEY_LOCAL_MACHINE\SOFTWARE\PhapSoftware\ImageGlass\Capabilities]
"ApplicationName"="ImageGlass"
"ApplicationDescription"="A lightweight, versatile image viewer"
"ApplicationIcon"="\"C:\\Program Files\\ImageGlass\\ImageGlass.exe\",0"

[HKEY_LOCAL_MACHINE\SOFTWARE\PhapSoftware\ImageGlass\Capabilities\FileAssociations]
".svg"="ImageGlass.AssocFile.SVG"
".jpg"="ImageGlass.AssocFile.JPG"
".jpeg"="ImageGlass.AssocFile.JPEG"
".gif"="ImageGlass.AssocFile.GIF"
".webp"="ImageGlass.AssocFile.WEBP"
".bmp"="ImageGlass.AssocFile.BMP"
".ico"="ImageGlass.AssocFile.ICO"
```

**ImageGlass.AssocFile.reg**
```
Windows Registry Editor Version 5.00

[HKEY_LOCAL_MACHINE\SOFTWARE\Classes\ImageGlass.AssocFile.SVG]
@="SVG file image"

[HKEY_LOCAL_MACHINE\SOFTWARE\Classes\ImageGlass.AssocFile.SVG\DefaultIcon]
@="\"C:\\Program Files\\ImageGlass\\ImageGlass.exe\",0"

[HKEY_LOCAL_MACHINE\SOFTWARE\Classes\ImageGlass.AssocFile.SVG\shell]

[HKEY_LOCAL_MACHINE\SOFTWARE\Classes\ImageGlass.AssocFile.SVG\shell\open]
"FriendlyAppName"="ImageGlass"

[HKEY_LOCAL_MACHINE\SOFTWARE\Classes\ImageGlass.AssocFile.SVG\shell\open\command]
@="\"C:\\Program Files\\ImageGlass\\ImageGlass.exe\" \"%1\""





[HKEY_LOCAL_MACHINE\SOFTWARE\Classes\ImageGlass.AssocFile.PNG]
@="PNG file image"

[HKEY_LOCAL_MACHINE\SOFTWARE\Classes\ImageGlass.AssocFile.PNG\DefaultIcon]
@="\"C:\\Program Files\\ImageGlass\\ImageGlass.exe\",0"

[HKEY_LOCAL_MACHINE\SOFTWARE\Classes\ImageGlass.AssocFile.PNG\shell]

[HKEY_LOCAL_MACHINE\SOFTWARE\Classes\ImageGlass.AssocFile.PNG\shell\open]
"FriendlyAppName"="ImageGlass"

[HKEY_LOCAL_MACHINE\SOFTWARE\Classes\ImageGlass.AssocFile.PNG\shell\open\command]
@="\"C:\\Program Files\\ImageGlass\\ImageGlass.exe\" \"%1\""





[HKEY_LOCAL_MACHINE\SOFTWARE\Classes\ImageGlass.AssocFile.JPG]
@="JPG file image"

[HKEY_LOCAL_MACHINE\SOFTWARE\Classes\ImageGlass.AssocFile.JPG\DefaultIcon]
@="\"C:\\Program Files\\ImageGlass\\ImageGlass.exe\",0"

[HKEY_LOCAL_MACHINE\SOFTWARE\Classes\ImageGlass.AssocFile.JPG\shell]

[HKEY_LOCAL_MACHINE\SOFTWARE\Classes\ImageGlass.AssocFile.JPG\shell\open]
"FriendlyAppName"="ImageGlass"

[HKEY_LOCAL_MACHINE\SOFTWARE\Classes\ImageGlass.AssocFile.JPG\shell\open\command]
@="\"C:\\Program Files\\ImageGlass\\ImageGlass.exe\" \"%1\""




[HKEY_LOCAL_MACHINE\SOFTWARE\Classes\ImageGlass.AssocFile.GIF]
@="GIF file image"

[HKEY_LOCAL_MACHINE\SOFTWARE\Classes\ImageGlass.AssocFile.GIF\DefaultIcon]
@="\"C:\\Program Files\\ImageGlass\\ImageGlass.exe\",0"

[HKEY_LOCAL_MACHINE\SOFTWARE\Classes\ImageGlass.AssocFile.GIF\shell]

[HKEY_LOCAL_MACHINE\SOFTWARE\Classes\ImageGlass.AssocFile.GIF\shell\open]
"FriendlyAppName"="ImageGlass"

[HKEY_LOCAL_MACHINE\SOFTWARE\Classes\ImageGlass.AssocFile.GIF\shell\open\command]
@="\"C:\\Program Files\\ImageGlass\\ImageGlass.exe\" \"%1\""




[HKEY_LOCAL_MACHINE\SOFTWARE\Classes\ImageGlass.AssocFile.WEBP]
@="WEBP file image"

[HKEY_LOCAL_MACHINE\SOFTWARE\Classes\ImageGlass.AssocFile.WEBP\DefaultIcon]
@="\"C:\\Program Files\\ImageGlass\\ImageGlass.exe\",0"

[HKEY_LOCAL_MACHINE\SOFTWARE\Classes\ImageGlass.AssocFile.WEBP\shell]

[HKEY_LOCAL_MACHINE\SOFTWARE\Classes\ImageGlass.AssocFile.WEBP\shell\open]
"FriendlyAppName"="ImageGlass"

[HKEY_LOCAL_MACHINE\SOFTWARE\Classes\ImageGlass.AssocFile.WEBP\shell\open\command]
@="\"C:\\Program Files\\ImageGlass\\ImageGlass.exe\" \"%1\""



[HKEY_LOCAL_MACHINE\SOFTWARE\Classes\ImageGlass.AssocFile.BMP]
@="BMP file image"

[HKEY_LOCAL_MACHINE\SOFTWARE\Classes\ImageGlass.AssocFile.BMP\DefaultIcon]
@="\"C:\\Program Files\\ImageGlass\\ImageGlass.exe\",0"

[HKEY_LOCAL_MACHINE\SOFTWARE\Classes\ImageGlass.AssocFile.BMP\shell]

[HKEY_LOCAL_MACHINE\SOFTWARE\Classes\ImageGlass.AssocFile.BMP\shell\open]
"FriendlyAppName"="ImageGlass"

[HKEY_LOCAL_MACHINE\SOFTWARE\Classes\ImageGlass.AssocFile.BMP\shell\open\command]
@="\"C:\\Program Files\\ImageGlass\\ImageGlass.exe\" \"%1\""



[HKEY_LOCAL_MACHINE\SOFTWARE\Classes\ImageGlass.AssocFile.ICO]
@="ICO file image"

[HKEY_LOCAL_MACHINE\SOFTWARE\Classes\ImageGlass.AssocFile.ICO\DefaultIcon]
@="\"C:\\Program Files\\ImageGlass\\ImageGlass.exe\",0"

[HKEY_LOCAL_MACHINE\SOFTWARE\Classes\ImageGlass.AssocFile.ICO\shell]

[HKEY_LOCAL_MACHINE\SOFTWARE\Classes\ImageGlass.AssocFile.ICO\shell\open]
"FriendlyAppName"="ImageGlass"

[HKEY_LOCAL_MACHINE\SOFTWARE\Classes\ImageGlass.AssocFile.ICO\shell\open\command]
@="\"C:\\Program Files\\ImageGlass\\ImageGlass.exe\" \"%1\""




[HKEY_LOCAL_MACHINE\SOFTWARE\Classes\ImageGlass.AssocFile.JPEG]
@="JPEG file image"

[HKEY_LOCAL_MACHINE\SOFTWARE\Classes\ImageGlass.AssocFile.JPEG\DefaultIcon]
@="\"C:\\Program Files\\ImageGlass\\ImageGlass.exe\",0"

[HKEY_LOCAL_MACHINE\SOFTWARE\Classes\ImageGlass.AssocFile.JPEG\shell]

[HKEY_LOCAL_MACHINE\SOFTWARE\Classes\ImageGlass.AssocFile.JPEG\shell\open]
"FriendlyAppName"="ImageGlass"

[HKEY_LOCAL_MACHINE\SOFTWARE\Classes\ImageGlass.AssocFile.JPEG\shell\open\command]
@="\"C:\\Program Files\\ImageGlass\\ImageGlass.exe\" \"%1\""
```



It's working in my machine.
![image](https://cloud.githubusercontent.com/assets/3154213/23666476/1b0fc7f4-0396-11e7-84b0-50f61cbebb4a.png)

**Comment by devcompl** [comment]
No, unfortunately it didn't work on my machine.
Maybe you have some leftover keys from previous installations or something like that.

**Comment by s8321414** [comment]
@d2phap I think you can remove version number from "default program" panel.

**Comment by d2phap** [comment]
Hi, I missed this:

**RegisteredApplications.reg**
```
Windows Registry Editor Version 5.00

[HKEY_LOCAL_MACHINE\SOFTWARE\RegisteredApplications]
"ImageGlass"="Software\\PhapSoftware\\ImageGlass\\Capabilities"
```

**Comment by devcompl** [comment]
@d2phap 
Seems to not work with windows 10 "Default apps".
But I could set ImageGlass as default from the old "Control Panel\All Control Panel Items\Default Programs\Set Default Programs".

Now in "Default apps" the "Photo viewer" app is empty.
![applicationframehost_2017-03-08_14-12-05](https://cloud.githubusercontent.com/assets/902882/23705283/59c9c2a6-0409-11e7-930e-acaccffd9293.png)

**Comment by d2phap** [comment]
Hi,

Please delete all ```ImageGlass.AssocFile.*``` keys in ```HKEY_LOCAL_MACHINE\SOFTWARE\Classes\```
Then run the below files again. I have installed new Windows OS to test it. It works as expected.



**RegisteredApplications.reg**
```
Windows Registry Editor Version 5.00

[HKEY_LOCAL_MACHINE\SOFTWARE\RegisteredApplications]
"ImageGlass"="Software\\PhapSoftware\\ImageGlass\\Capabilities"
```

**ImageGlass.AssocFile.reg.reg**
```
Windows Registry Editor Version 5.00

[HKEY_LOCAL_MACHINE\SOFTWARE\Classes\ImageGlass.AssocFile]
@="ImageGlass file image"

[HKEY_LOCAL_MACHINE\SOFTWARE\Classes\ImageGlass.AssocFile\DefaultIcon]
@="\"C:\\Program Files\\ImageGlass\\ImageGlass.exe\",0"

[HKEY_LOCAL_MACHINE\SOFTWARE\Classes\ImageGlass.AssocFile\shell]

[HKEY_LOCAL_MACHINE\SOFTWARE\Classes\ImageGlass.AssocFile\shell\open]
"FriendlyAppName"="ImageGlass"

[HKEY_LOCAL_MACHINE\SOFTWARE\Classes\ImageGlass.AssocFile\shell\open\command]
@="\"C:\\Program Files\\ImageGlass\\ImageGlass.exe\" \"%1\""
```

**ImageGlass.Capabilities.reg**
```
Windows Registry Editor Version 5.00

[HKEY_LOCAL_MACHINE\SOFTWARE\PhapSoftware]

[HKEY_LOCAL_MACHINE\SOFTWARE\PhapSoftware\ImageGlass]

[HKEY_LOCAL_MACHINE\SOFTWARE\PhapSoftware\ImageGlass\Capabilities]
"ApplicationName"="ImageGlass"
"ApplicationDescription"="A lightweight, versatile image viewer"
"ApplicationIcon"="\"C:\\Program Files\\ImageGlass\\ImageGlass.exe\",0"

[HKEY_LOCAL_MACHINE\SOFTWARE\PhapSoftware\ImageGlass\Capabilities\FileAssociations]
".svg"="ImageGlass.AssocFile"
".webp"="ImageGlass.AssocFile"
".bmp"="ImageGlass.AssocFile"
".dib"="ImageGlass.AssocFile"
".jpg"="ImageGlass.AssocFile"
".jpeg"="ImageGlass.AssocFile"
".jpe"="ImageGlass.AssocFile"
".jfif"="ImageGlass.AssocFile"
".gif"="ImageGlass.AssocFile"
".tif"="ImageGlass.AssocFile"
".tiff"="ImageGlass.AssocFile"
".png"="ImageGlass.AssocFile"
".ico"="ImageGlass.AssocFile"
```

**Comment by devcompl** [comment]
Yes now it works perfect.
Maybe one thing to tweak now would be to remove the version number from the name in "Default apps" like @s8321414 suggested.

Thanks!

**Comment by d2phap** [comment]
Thanks for your reply 👍

**Comment by s8321414** [comment]
❤️

**Comment by d2phap** [comment]
The issue is fixed (same as #16).
This is a tesing version: http://www.imageglass.org/upload/moon/ImageGlass_3.x_17032017.zip
